src/librustc_typeck/check/method/probe.rs

```diff
@@ -128,7 +133,7 @@ pub enum PickKind<'tcx> {
                     ty::PolyTraitRef<'tcx>),
 }
 
-pub type PickResult<'tcx> = Result<Pick<'tcx>, MethodError<'tcx>>;
+pub type PickResult<'tcx> = Result<Vec<Pick<'tcx>>, MethodError<'tcx>>;
```

Comment:
We should definitely leave a comment here as to what is going on in this type (why a vector? under what conditions are errors reported?). I think that the `Ok` case represents a set of possible picks -- if a given method name is used, this set is guaranteed to be of cardinality 1, right? Otherwise, is it guaranteed to be non-empty?

I'm trying to decide if this is the best type -- perhaps it is, but it seems strange to me that we can be searching for many methods at once and yet possibly get an error. I have to remind myself under what conditions a `MethodError` is a valid result.

I am mostly concerned because later on we have those `Option<PickResult>` and there is a lot of kind of awkward bridging code. I do wonder if `PickResult` could stay the same, but `Option<PickResult>` becomes `Vec<PickResult>` instead.